alibi_detect/utils/schemas.py

```diff
@@ -0,0 +1,431 @@
+    learning_rate: float = 1e-3
+    batch_size: int = 32
+    preprocess_batch_fn: Optional[str] = None
+    epochs: int = 3
+    verbose: int = 0
+    train_kwargs: Optional[dict] = None
+    dataset: str = '@alibi_detect.utils.tensorflow.data.TFDataset'
+    device: Literal['cpu', 'cuda', None] = None
+
+
+class SpotTheDiffDriftConfig(DriftDetectorConfig):
+    """
+    Unresolved schema for SpotTheDiffDrift detector.
+    """
+    binarize_preds: bool = False
+    train_size: Optional[float] = .75
+    n_folds: Optional[int] = None
+    retrain_from_scratch: bool = True
+    seed: int = 0
+    optimizer: Union[str, dict, None] = None  # dict as can pass dict to tf.keras.optimizers.deserialize
+    learning_rate: float = 1e-3
+    batch_size: int = 32
+    preprocess_batch_fn: Optional[str] = None
+    epochs: int = 3
+    verbose: int = 0
+    train_kwargs: Optional[dict] = None
+    dataset: str = '@alibi_detect.utils.tensorflow.data.TFDataset'
+    kernel: Union[str, KernelConfig, None] = None
+    n_diffs: int = 1
+    initial_diffs: Optional[str] = None
+    l1_reg: float = 0.01
+    device: Literal['cpu', 'cuda', None] = None
+
+
+class LearnedKernelDriftConfig(DriftDetectorConfig):
+    """
+    Unresolved schema for LearnedKernelDrift detector.
+    """
+    kernel: Union[str, DeepKernelConfig]
+    preprocess_at_init: bool = True
+    update_x_ref: Optional[Dict[str, int]] = None
+    n_permutations: int = 100
+    var_reg: float = 1e-5
+    reg_loss_fn: Optional[str] = None
+    train_size: Optional[float] = .75
+    retrain_from_scratch: bool = True
+    optimizer: Union[str, dict, None] = None  # dict as can pass dict to tf.keras.optimizers.deserialize
+    learning_rate: float = 1e-3
+    batch_size: int = 32
+    preprocess_batch_fn: Optional[str] = None
+    epochs: int = 3
+    verbose: int = 0
+    train_kwargs: Optional[dict] = None
+    dataset: str = '@alibi_detect.utils.tensorflow.data.TFDataset'
+    device: Literal['cpu', 'cuda', None] = None
+
+
+class KSDriftConfigResolved(DriftDetectorConfigResolved, KSDriftConfig):
+    """
+    Resolved schema for KSDrift detector.
+    """
+    pass
+
+
+class ChiSquareDriftConfigResolved(DriftDetectorConfigResolved, ChiSquareDriftConfig):
+    """
+    Resolved schema for ChiSquareDrift detector.
+    """
+    pass
+
+
+class TabularDriftConfigResolved(DriftDetectorConfigResolved, TabularDriftConfig):
+    """
+    Resolved schema for TabularDrift detector.
+    """
+    pass
+
+
+class CVMDriftConfigResolved(DriftDetectorConfigResolved, CVMDriftConfig):
+    pass
+
+
+class FETDriftConfigResolved(DriftDetectorConfigResolved, FETDriftConfig):
+    pass
+
+
+class MMDDriftConfigResolved(DriftDetectorConfigResolved, MMDDriftConfig):
+    """
+    Resolved schema for MMDDrift detector.
+    """
+    kernel: Union[Callable, KernelConfigResolved, None]
+    sigma: Optional[np.ndarray]
```

Comment:
I'm assuming there is no default value because the default should come from one of the parent models?